usl_pipeline/cloud_functions/main.py

```diff
@@ -603,9 +607,10 @@ def _build_feature_matrix_from_archive(
 
 
 def _build_wps_feature_matrix(fd: IO[bytes]) -> Tuple[NDArray, FeatureMetadata]:
+    fd_bytes = io.BytesIO(fd.read())
     # Ignore type checker error - BytesIO inherits from expected type BufferedIOBase
     # https://shorturl.at/lk4om
-    with xarray.open_dataset(fd) as ds:  # type: ignore
+    with xarray.open_dataset(fd_bytes) as ds:  # type: ignore
```

Comment:
Why pass `fd.read()` rather than `fd` here? My hope would be that `open_dataset` can stream the data rather than read it all in at once.

Reply:
yeah looks like we can do it if we specify the right engine. updated. tested locally and this should work!